Source/JavaScriptCore/runtime/WeakMapPrototype.cpp

```diff
@@ -96,7 +98,9 @@ JSC_DEFINE_HOST_FUNCTION(protoFuncWeakMapHas, (JSGlobalObject* globalObject, Cal
     if (!map)
         return JSValue::encode(jsUndefined());
     JSValue key = callFrame->argument(0);
-    return JSValue::encode(jsBoolean(key.isObject() && map->has(asObject(key))));
+    if (UNLIKELY(!canBeHeldWeakly(key)))
+        return JSValue::encode(jsBoolean(false));
```

Comment:
ditto. Probably this happens in other access functions too, but I won't annotate them all

Reply:
Changed.